exec/cnex/module.c

```diff
@@ -0,0 +1,52 @@
+#include "module.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#include "bytecode.h"
+#include "cell.h"
+#include "exec.h"
+
+
+TModule *module_newModule(const char *name)
+{
+    TModule *r = malloc(sizeof(TModule));
+    if (r == NULL) {
+        fatal_error("Could not allocate memory for new module: %s", name);
+    }
+    memset(r, 0x00, sizeof(TModule));
+
+    r->name = malloc(strlen(name) + 1);
+    if (r->name == NULL) {
+        fatal_error("Could not allocate memory for module name: %s", name);
+    }
+    strcpy(r->name, name);
+
+    r->bytecode = bytecode_newBytecode();
+
```

Comment:
There are a couple of structure members (`code`, `globals`) that are uninitialized here and yet freed in the `module_freeModule` function.

Reply:
Those are allocated elsewhere.  I'm re-working the implementation, however.

Reply:
Fixed.